openassessment/workflow/models.py

```diff
@@ -364,11 +411,24 @@ def set_score(self, score):
                 'points_possible'.
 
         """
-        sub_api.set_score(
-            self.submission_uuid,
-            score["points_earned"],
-            score["points_possible"]
-        )
+        if not self.staff_score_exists():
+            sub_api.set_score(
+                self.submission_uuid,
+                score["points_earned"],
+                score["points_possible"]
+            )
+
+    def staff_score_exists(self):
```

Comment:
Don't forget a docstring for this. I'm also wondering if there's anyway to clean up all the `None`s we check here.